services/src/handlers/model_training.rs

```diff
@@ -0,0 +1,271 @@
+
+        let qry: QueryRectangle<BoundingBox2D> = VectorQueryRectangle {
+            spatial_bounds,
+            time_interval,
+            spatial_resolution,
+        };
+
+        let xg_train = crate::machine_learning::MLTrainRequest {
+            query: qry,
+            params: crate::machine_learning::XgboostTrainingParams {
+                no_data_value: -1_000.,
+                training_config,
+                feature_names: vec![Some("a".into()), Some("b".into())],
+                label_name: Some("target".into()),
+                aggregate_variant: crate::machine_learning::MachineLearningAggregator::Simple,
+                memory_limit: Some(1024),
+            },
+            input_workflows: vec![workflow_a, workflow_b],
+            label_workflow: vec![workflow_target],
+        };
+
+        let req = test::TestRequest::post()
+            .uri("/ml/train")
+            .append_header((header::AUTHORIZATION, Bearer::new(session_id.to_string())))
+            .set_json(xg_train);
+
+        let res = send_test_request(req, app_ctx.clone()).await;
+
+        assert_eq!(res.status(), 200, "{:?}", res.response());
+
+        let task_response =
+            serde_json::from_str::<TaskResponse>(&read_body_string(res).await).unwrap();
+
+        let tasks = Arc::new(ctx.tasks());
+
+        wait_for_task_to_finish(tasks.clone(), task_response.task_id).await;
+
+        let status = tasks.get_task_status(task_response.task_id).await.unwrap();
+
+        let response = if let TaskStatus::Completed { info, .. } = status {
+            info.as_any_arc()
+                .downcast::<MachineLearningModelFromWorkflowResult>()
+                .unwrap()
+                .as_ref()
+                .clone()
+        } else {
+            panic!("Task must be completed");
+        };
+
+        let model = response.model;
+        let model_uuid = response.model_uuid;
+
+        // get the content of the test model on disk to compare against
+        let test_model_path = PathBuf::from(geoengine_datatypes::test_data!(
+            "pro/ml/xgboost/test_model.json"
+        ));
+        let mut test_model_bytes: Vec<u8> = Vec::new();
+        let mut f = File::open(test_model_path)
+            .await
+            .expect("Unable to open file of test model");
+        f.read_to_end(&mut test_model_bytes)
+            .await
+            .expect("Could not read file of test model");
+
+        // check that the returned model is as expected
+        assert_eq!(&test_model_bytes as &[u8], model.to_string().as_bytes());
+
+        // also check, that the model (which was written after training) on disk is as expected
+        let model_path = &cfg.model_defs_path.join(model_uuid);
+
+        let exe_ctx = ctx.execution_context().unwrap();
+        let model_from_disk = exe_ctx
+            .read_ml_model(model_path.clone())
+            .await
+            .expect("Could not read specified ml model from disk");
```

Comment:
why are the featue_names Options if they lead to an error here? You could already fail earlier.